src/utils/Plotter.cpp

```diff
@@ -100,236 +63,336 @@ template <int D> void Plotter<D>::setSuffix(int t, const std::string &s) {
     this->suffix.insert(std::pair<int, std::string>(t, s));
 }
 
-/** Parametric plot of a function
+/** Grid plot of a MWTree
 
-    Plots the function func parametrically between the endpoints A and B,
-    to a file named fname + file extension (".line" as default). Endpoints and
-    the number of points must have been set at this point (default values are
-    1000 points between the boundaries of the function).
+    Writes a file named fname + file extension (".grid" as default) to be read
+    by geomview to visualize the grid (of endNodes)	where the multiresolution
+    function is defined. In MPI, each process will write a separate file, and
+    will print only	nodes owned by itself (pluss the rootNodes).
 */
-template <int D> void Plotter<D>::linePlot(const RepresentableFunction<D> &func, const std::string &fname) {
-    println(20, "----------Line Plot-----------");
-    if (not verifyRange()) MSG_ERROR("Zero range");
-    calcLineCoordinates();
-    evaluateFunction(func);
+template <int D> void Plotter<D>::gridPlot(const MWTree<D> &tree, const std::string &fname) {
+    println(20, "----------Grid Plot-----------");
     std::stringstream file;
-    file << fname << this->suffix[Plotter<D>::Line];
+    file << fname << this->suffix[Plotter<D>::Grid];
     openPlot(file.str());
-    writeLineData();
+    writeGrid(tree);
     closePlot();
     printout(20, std::endl);
 }
 
-/** Surface plot of a function
+/** Parametric plot of a function
 
-    Plots the function func in 2D between the endpoints A and B, to a file
-    named fname + file extension (".surf" as default). Endpoints and
-    the number of points must have been set at this point (default values are
-    1000 points between the boundaries of the function). The nPoints points
-    will be distributed evenly in the two dimensions.
+    Plots the function func parametrically along the vector A starting
+    from the origin O to a file named fname + file extension (".line"
+    as default).
 */
-template <int D> void Plotter<D>::surfPlot(const RepresentableFunction<D> &func, const std::string &fname) {
-    println(20, "--------Surface Plot----------");
-    if (not verifyRange()) MSG_ERROR("Zero range");
-    calcSurfCoordinates();
-    evaluateFunction(func);
+template <int D>
```

Comment:
I think you can unify the two `linePlot` function using a template template parameter to abstract the second argument (`RepresentableFunction` vs `FunctionTree`).

Reply:
Hmm, the difference between the two is actually `const` vs non-`const`. `FunctionTree` is a sub-class of `RepresentableFunction`, but for technical reasons it is not `const` upon function evaluation. I will have a go at `const`-ifying it, as it conceptually should be.

Reply:
Ugh, I totally missed the `const`-ness. I probably should refrain from reviewing at night... Anyway, it's still probably worthwhile using a template template, in case this function ends up being used with parameters that are not in the `RepresentableFunction` hierarchy.

Reply:
I managed to `const`-ify the `FunctionTree::evalf` so the `FunctionTree` instance is no longer necessary.